name=source-event
connector.class=io.confluent.connect.jdbc.JdbcSourceConnector
tasks.max=1
connection.url=jdbc:postgresql://localhost/source?user=postgres&password=password
mode=timestamp+incrementing
# mode=timestamp
timestamp.column.name=event_date
incrementing.column.name=event_id
topic.prefix=profile-
table.whitelist=event_data
# poll.interval.ms=1000

|    validate.non.null = true
connect            |  (io.confluent.connect.jdbc.source.JdbcSourceTaskConfig)
connect            | [2019-01-20 02:29:47,128] INFO After filtering the tables are: "public"."sensor_reading" (io.confluent.connect.jdbc.source.TableMonitorThread)
connect            | [2019-01-20 02:29:47,133] INFO Finished starting connectors and tasks (org.apache.kafka.connect.runtime.distributed.DistributedHerder)
connect            | [2019-01-20 02:29:47,650] INFO WorkerSourceTask{id=sensor-source-0} Committing offsets (org.apache.kafka.connect.runtime.WorkerSourceTask)
connect            | [2019-01-20 02:29:47,652] INFO WorkerSourceTask{id=sensor-source-0} flushing 0 outstanding messages for offset commit (org.apache.kafka.connect.runtime.WorkerSourceTask)
connect            | [2019-01-20 02:29:47,652] ERROR WorkerSourceTask{id=sensor-source-0} Task threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerTask)
connect            | org.apache.kafka.connect.errors.ConnectException: Cannot make incremental queries using timestamp columns [reading_timestamp] on "public"."sensor_reading" because all of these columns nullable.
connect            |    at io.confluent.connect.jdbc.source.JdbcSourceTask.validateNonNullable(JdbcSourceTask.java:379)
connect            |    at io.confluent.connect.jdbc.source.JdbcSourceTask.start(JdbcSourceTask.java:154)
connect            |    at org.apache.kafka.connect.runtime.WorkerSourceTask.execute(WorkerSourceTask.java:198)
connect            |    at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:175)
connect            |    at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:219)
connect            |    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
connect            |    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
connect            |    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
connect            |    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
connect            |    at java.lang.Thread.run(Thread.java:748)
connect            | [2019-01-20 02:29:47,654] ERROR WorkerSourceTask{id=sensor-source-0} Task is being killed and will not recover until manually restarted (org.apa